libs/sql-schema-describer/src/mssql.rs

```diff
@@ -0,0 +1,570 @@
+use super::*;
+use once_cell::sync::Lazy;
+use quaint::prelude::Queryable;
+use regex::Regex;
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    convert::TryInto,
+    sync::Arc,
+};
+
+static DEFAULT_INT: Lazy<Regex> = Lazy::new(|| Regex::new(r"\(\((.*)\)\)").unwrap());
+static DEFAULT_NON_INT: Lazy<Regex> = Lazy::new(|| Regex::new(r"\((.*)\)").unwrap());
+
+pub struct SqlSchemaDescriber {
+    conn: Arc<dyn Queryable + Send + Sync + 'static>,
+}
+
+#[async_trait::async_trait]
+impl super::SqlSchemaDescriberBackend for SqlSchemaDescriber {
+    async fn list_databases(&self) -> crate::SqlSchemaDescriberResult<Vec<String>> {
+        Ok(self.get_databases().await)
+    }
+
+    async fn get_metadata(&self, schema: &str) -> crate::SqlSchemaDescriberResult<SQLMetadata> {
+        let table_count = self.get_table_names(schema).await.len();
+        let size_in_bytes = self.get_size(schema).await;
+
+        Ok(SQLMetadata {
+            table_count,
+            size_in_bytes,
+        })
+    }
+
+    async fn describe(&self, schema: &str) -> crate::SqlSchemaDescriberResult<crate::SqlSchema> {
+        debug!("describing schema '{}'", schema);
+
+        let mut columns = self.get_all_columns(schema).await;
+        let mut indexes = self.get_all_indices(schema).await;
+        let mut foreign_keys = self.get_foreign_keys(schema).await;
+
+        let table_names = self.get_table_names(schema).await;
+        let mut tables = Vec::with_capacity(table_names.len());
+
+        for table_name in table_names {
+            let table = self.get_table(&table_name, &mut columns, &mut indexes, &mut foreign_keys);
+            tables.push(table);
+        }
+
+        Ok(SqlSchema {
+            tables,
+            enums: vec![],
```

Comment:
Question because I really don't know: does MSSQL have enums?

Reply:
It does not.